src/main/java/com/iexec/worker/compute/ComputeManagerService.java

```diff
@@ -74,8 +100,18 @@ public boolean downloadApp(TaskDescription taskDescription) {
         if (!isDockerType || taskDescription.getAppUri() == null) {
             return false;
         }
+
+        final long maxExecutionTime = taskDescription.getMaxExecutionTime();
+        final long pullTimeout = Math.max(
```

Comment:
Why not moving this calculation block to a dedicated method in order to easily unit test it?

Reply:
Sure, good idea!